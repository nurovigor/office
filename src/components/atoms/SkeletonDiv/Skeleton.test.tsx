import React from 'react';
import { render, screen } from '@testing-library/react';
import { SkeletonDiv } from './';

describe('Preloader component', () => {
	it('renders Preloader component', () => {
		const { asFragment } = render(<SkeletonDiv height={50} width={50} />);
		expect(asFragment()).toMatchSnapshot();
	});
	it('renders with percent width', () => {
		render(<SkeletonDiv height={50} percentWidth={50} />);
		expect(screen.getByTestId('skeleton')).toHaveStyle({ minWidth: '50%' });
	});

	it('renders with px width', () => {
		render(<SkeletonDiv height={50} width={50} />);
		expect(screen.getByTestId('skeleton')).toHaveStyle({ minWidth: '50px' });
	});
});
