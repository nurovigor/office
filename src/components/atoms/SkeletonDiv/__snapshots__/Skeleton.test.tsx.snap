// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component renders Preloader component 1`] = `
<DocumentFragment>
  <div
    data-testid="skeleton"
    style="min-width: 50px; height: 50px;"
  />
</DocumentFragment>
`;
